cli/cmd/generate_aws.go

```diff
@@ -0,0 +1,399 @@
+	QuestionConsolidatedCloudtrail      = "Use consolidated Cloudtrail?"
+	QuestionUseExistingCloudtrail       = "Use an existing Cloudtrail?"
+	QuestionCloudtrailExistingBucketArn = "Specify an existing bucket ARN used for Cloudtrail logs:"
+	QuestionForceDestroyS3Bucket        = "Should the new S3 bucket have force destroy enabled?"
+	QuestionExistingIamRoleName         = "Specify an existing IAM role name for Cloudtrail access"
+	QuestionExistingIamRoleArn          = "Specify an existing IAM role ARN for Cloudtrail access"
+	QuestionExistingIamRoleExtId        = "Specify the external ID to be used with the existing IAM role"
+	QuestionPrimaryAwsAccountProfile    = "Before adding subaccounts, your primary AWS account profile name must be set; which profile should the main account use?"
+	QuestionSubAccountProfileName       = "Supply the profile name for this additional AWS account:"
+	QuestionSubAccountRegion            = "What region should be used for this account?"
+	QuestionSubAccountAddMore           = "Add another AWS account?"
+	QuestionSubAccountReplace           = "Currently configured AWS subaccounts: %s, replace?"
+	QuestionConfigAdvanced              = "Configure advanced integration options?"
+	QuestionAnotherAdvancedOpt          = "Configure another advanced integration option"
+	QuestionCustomizeOutputLocation     = "Provide the location for the output to be written:"
+
+	// select options
+	AdvancedOptDone        = "Done"
+	AdvancedOptCloudTrail  = "Additional Cloudtrail options"
+	AdvancedOptIamRole     = "Configure Lacework integration with an existing IAM role"
+	AdvancedOptAwsAccounts = "Add additional AWS Accounts to Lacework"
+	AdvancedOptLocation    = "Customize output location"
+
+	// original source: https://regex101.com/r/pOfxYN/1
+	AwsArnRegex = `^arn:(?P<Partition>[^:\n]*):(?P<Service>[^:\n]*):(?P<Region>[^:\n]*):(?P<AccountID>[^:\n]*):(?P<Ignore>(?P<ResourceType>[^:\/\n]*)[:\/])?(?P<Resource>.*)$`
+	// regex used for validating region input; note intentionally does not match gov cloud
+	AwsRegionRegex  = `(us|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d`
+	AwsProfileRegex = `([A-Za-z_0-9-]+)`
+)
+
+// create survey.Validator for string with regex
+func validateStringWithRegex(val interface{}, regex string, errorString string) error {
+	// the reflect value of the result
+	value := reflect.ValueOf(val)
+
+	// if the value passed is not a string
+	if value.Kind() != reflect.String {
+		return errors.New("value must be a string")
+	}
+
+	// if value doesn't match regex, return invalid arn
+	ok, err := regexp.MatchString(regex, value.String())
+	if err != nil {
+		return errors.Wrap(err, "failed to validate input")
+	}
+
+	if !ok {
+		return errors.New(errorString)
+	}
+
+	return nil
+}
+
+// survey.Validator for aws ARNs
+//
+// This isn't service/type specific but rather just validates that an ARN was entered that matches valid ARN formats
```

Comment:
nitpick: My IDE complains about top-level function comments that aren't prefixed with the function name. I can't claim to know if this is the standard practice in Go, so feel free to ignore.
IDE error:
`Comment should have the following format '<func_name>...' (with an optional leading article) `

Reply:
@rmoles 😬  yuk.  what IDE is that, intellij?

Reply:
As discussed on slack, it seems it is actually the IntelliJ IDE Go `Inspector` settings flagging this. This check does point to the [GoLang Docs](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences), however seems to go above and beyond golangci linter.
![image (4)](https://user-images.githubusercontent.com/15362434/145241289-c061a8b6-0d7b-48c9-8bb5-e6de758dbac4.png)

Reply:
This comment is definitely non blocking

Reply:
Just realising that this comment is on the wrong file the issue was showing in `lwgenerate/aws/aws.go`